python/lsst/daf/butler/_butler.py

```diff
@@ -371,7 +376,10 @@ def makeRepo(root: str, config: Union[Config, str, None] = None, standalone: boo
         config.dumpToUri(configURI, overwrite=overwrite)
 
         # Create Registry and populate tables
-        Registry.fromConfig(config, create=createRegistry, butlerRoot=root)
+        if createRegistry:
+            registryConfig = RegistryConfig(config.get("registry"))
```

Comment:
Doesn't this `config.get` subvert what `ConfigSubset` is trying to work out for itself? By this point it's a full ButlerConfig config so it shouldn't need to be given the "registry" key. We've also already created a RegistryConfig above so this may not be necessary.

Reply:
I believe that if `standalone` is false then `config` does not need to have all individual sub-keys. If "registry" key is missing from `config` then `RegistryConfig(config)` works differently - it loads defaults from `registry.yaml` but then also merges `config` contents, which I think is wrong. `RegistryConfig(config.get("registry"))` should be safer in that respect - if "registry" is missing then it does not try to merge `config`, only loads defaults.